### ISSUE TITLE
Add ability to disable the path rewriting

### ISSUE THREAD

**Issue by timsuchanek** [zeit/ncc#390]
This code:
```ts
const runtimePath = path.join(__dirname, '../runtime')
```

Is being incorrectly rewritten by ncc to
```ts
const runtimePath = __dirname + '/runtime'
```

I actually want it to stay as it is.

My setup looks like this:
`./scripts/download.js`
```ts
const { ensureBinaries } = require('@prisma/fetch-engine')
const path = require('path')
const debug = require('debug')('download')
const runtimePath = path.join(__dirname, '../runtime')
debug(`Downloading binaries to ${runtimePath}`)
ensureBinaries(runtimePath)
```

ncc command:
```ts
ncc build scripts/download.js -o download
```

Also it takes all files from the `../runtime` folder and copies them to `./download/runtime`, which I don't need and then manually have to remove.
Is there a way to disable it?

I was thinking of some hint that I could give ncc, maybe a comment in the code like this:
`// ncc-ignore-next-line`

I saw that you're doing [`eval` in the ncc source code](https://github.com/zeit/ncc/blob/654fdffb8d00b8c40b2d6e23de61b1ec9469010f/src/index.js#L32), is that also because of that problem?

**Comment by guybedford** [comment]
@timsuchanek yes some way to opt-out sounds sensible for these kinds of valid scenarios. Perhaps we could provide a list of directories to never emit like `ncc --noemit ../runtime --noemit ../other` or similar.

What I would suggest first is to try this with the latest ncc beta if you can as that already includes some changes to this emission, which might possibly have resolved this.

**Comment by guybedford** [comment]
And yes, `eval()` can be used as a workaround to opt-out of the analysis too.

**Comment by roryabraham** [comment]
I ran into a very similar scenario, and I think `// ncc-ignore-next-line` would be preferable to `--noemit folder` because it's a bit more flexible. I have a central build script that runs `ncc` for all my GitHub Actions, and maybe I'd want to ignore a directory in one action but not another.

**Comment by roryabraham** [comment]
I ran into this again, trying to bundle a Node script that was using `__dirname`, but now I understand the problem better.

- If `__dirname` is evaluated at compile time, then it will be an absolute path on whatever computer the bundled script was compiled on. That's not useful.
- If `__dirname` is evaluated at runtime, then it will resolve relative to wherever the bundled script ends up. This is different from how you'd expect it to work in a Node environment. In Node, if you import a mode that uses `__dirname` internally, the internal usage of `__dirname` is going to resolve to an absolute path of that module. Once it's bundled with ncc, it just refers to the dirname where the ncc-bundled script lives.

I also don't think this is a problem ncc can solve, because it should make no assumptions that the filesystem in which a script is _compiled_ will have any relevance to where it's run.

---

I understand it doesn't work in every case, but in my case, I was able to get the root path of the repo my script is running in:

```ts
function getRepoRoot(): string {
    return execSync('git rev-parse --show-toplevel', {
        encoding: 'utf8',
    }).trim();
}
```

Then use absolute paths resolved from that 🙂
